BestPractices.md

```diff
@@ -80,8 +80,53 @@ This chapter outlines best practices to use for developing in iOS and Android.
 
 ## Android
 
-### Common libraries
+### Architectures
+
+### Development Advices
 
-#### - Networking - Parsing
-#### - Image Download
-#### - Dependecy Injection
+* Try not to use nested fragments in activities.
+* Do not put too much code in activities.
+* Code blocks that are used more than one activities should put into *Utility* classes.
+* Rules of code guidelines must be followed while writing the code. 
+  * [**[*Java*]** Code Guideline](https://github.com/ahmetmentes/android-guidelines)
+  * [**[*Kotlin*]** Code Guideline](https://android.github.io/kotlin-guides/style.html)
+* Simple values that needs to be persisted in application can be put into SharedPreferences.
+* It's a good idea to make custom view attributes changeable from XML file.
+* *Do not use* hardcoded strings in layout and put all strings to **strings.xml** file.
+* All colors must be declared in **colors.xml** file to use.
+* Strings that are declared in **strings.xml** should be not in all caps. If you need to use in all caps, use textAllCaps feature.
```

Comment:
Why? If only all caps version is used in the whole app then you can write it as all caps. imho.

Reply:
You might need that text in the future as not all in caps. Of course, It's a preference not a must. and also, first letter capitalized strings are easier while translating as well. 
Should I delete this statement?

Reply:
Okay it makes sense but we can improve this like "If the exact same text has more than one version with different caps try to use textAllCaps to reduce the number of defined strings "